frontend/src/data/actions.js

```diff
@@ -17,12 +22,23 @@ export const singleRequest = {
   error: () => ({
     type: types.ERROR,
   }),
-  addUser: ({ name }) => ({
-    type: types.ADDUSER,
-    payload: { name },
+  createUser: ({ email, password }) => ({
+    type: types.CREATEUSER,
+    payload: { email, password },
   }),
-  downloadImage: ({ imageURL }) => ({
-    type: types.DOWNLOADIMAGE,
-    payload: { imageURL },
+  updateEmail: ({ email }) => ({
+    type: types.UPDATEEMAIL,
+    payload: { email },
+  }),
+  updatePassword: ({ password }) => ({
+    type: types.UPDATEPASSWORD,
+    payload: { password },
+  }),
+  signIn: ({ email, password }) => ({
+    type: types.SIGNIN,
+    payload: { email, password },
+  }),
+  authenticationSuccess: () => ({
```

Comment:
So that there will be another const for authenticate where you can combine put signin inside as for success action